presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/execution/NoopSqlFunctionExecutionModule.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.functionNamespace.execution;
+
+import com.facebook.presto.spi.function.SqlFunctionExecutor;
+import com.google.inject.Binder;
+import com.google.inject.Scopes;
+
+public class NoopSqlFunctionExecutionModule
```

Comment:
The non built-in UDFs are managed by extensions of class `AbstractSqlInvokedFunctionNamespaceManager`. In the PR, the Json file based function manager also extends the `AbstractSqlInvokedFunctionNamespaceManager`.

The `AbstractSqlInvokedFunctionNamespaceManager` class https://github.com/prestodb/presto/blob/19a9bb8a516ced26c8528f50b5d3f4cfa9d245a1/presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/AbstractSqlInvokedFunctionNamespaceManager.java#L77-L80 has a `SqlFunctionExecutors` member, which is used for execution of remote UDFs (and in our case, the CPP UDF will be executed like other built-in UDFs in presto worker, we will put a noop instance of `SqlFunctionExecutors` here).

The `SqlFunctionExecutors` https://github.com/prestodb/presto/blob/19a9bb8a516ced26c8528f50b5d3f4cfa9d245a1/presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/execution/SqlFunctionExecutors.java#L42-L47 has an Inject constructor, and it has a `SqlFunctionExecutor` to do the real execution.

In summary, it's `AbstractSqlInvokedFunctionNamespaceManager` -> `SqlFunctionExecutors` -> `SqlFunctionExecutor`.

Currently, for the thrift implementation of `MySqlFunctionNamespaceManager`, it has `SimpleAddressSqlFunctionExecutorsModule` which corresponds to the `SqlFunctionExecutors` class. And it has `SimpleAddressThriftSqlFunctionExecutionModule` corresponds to the `SqlFunctionExecutor`.

In this PR, the `NoopSqlFunctionExecutorsModule` corresponds to `SqlFunctionExecutors` and `NoopSqlFunctionExecutionModule` corresponds to `SqlFunctionExecutor`.

In our case, since we currently only intends to support CPP UDF for the Json function manager, and we do not rely on the `SqlFunctionExecutor` for execution. We can simply create a noop instance of `SqlFunctionExecutors`, similar to what we have here https://github.com/prestodb/presto/blob/19a9bb8a516ced26c8528f50b5d3f4cfa9d245a1/presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestRewriteFilterWithExternalFunctionToProject.java#L69-L76.

However, the drawback is that we will hard code the supportedLanguages for `SqlFunctionExecutors`.

We can simplify and has less dependency injection if we do not want to keep the flexibility. Either way sounds fine to me. What do you think?

Reply:
Thanks for the explanation.  I don't have strong feelings, but think it's okay as is, so let's leave it.